src/GradingStudents/GradingStudents.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\GradingStudents;
+
+use InvalidArgumentException;
+
+class GradingStudents
+{
+    public function roundGrades($grades)
+    {
+        $grades = explode(PHP_EOL, $grades);
+        $students = (int) array_shift($grades);
+        $this->validateInput($students, $grades);
+
+        return implode(PHP_EOL, array_map(function ($grade) {
+            if ($grade < 0 || $grade > 100) {
```

Comment:
Você pode evitar executar este IF várias vezes ao validar apenas uma vez antes do seu loop (array_map).

Você pode utilizar as funções `min()` e `max()` para obter o menor e maior valores neste array. Se `min($grades)` for menor que 0 ou `max($grades)` for maior que 100, você lança a exceção.

Refs:
- https://www.php.net/manual/en/function.min.php
- https://www.php.net/manual/en/function.max